### PR TITLE
Add unary default value and validation for max_beam_width configuration

### PR DESCRIPTION
**Pull request by joostinyi**
## :rocket: What

- Adds unary default value and field validator to `trt_llm.build.max_beam_width` as support is limited to this case for the time being.
- Bumps truss version for briton requirements to leverage this config change.

## :microscope: Testing
- Added unit test for field validation

### CODE REVIEW

**Review by bdubayah — APPROVED** [review]
Validation looks good to me. But if we constraining max_beam_width = 1, should we just remove it from the config and docs?

File: truss/config/trt_llm.py [hunk 1]
```diff
@@ -70,6 +70,14 @@ class TrussTRTLLMBuildConfiguration(BaseModel):
     kv_cache_free_gpu_mem_fraction: float = 0.9
     num_builder_gpus: Optional[int] = None
 
+    @field_validator("max_beam_width", mode="after")
+    @classmethod
+    def check_max_beam_width(cls, v: int):
+        if isinstance(v, int):
+            if v != 1:
+                raise ValueError("max_beam_width must be unary")
```
Comment (aspctu) [inline truss/config/trt_llm.py:78]: This error can be simpler: 

```
Only `max_beam_width` equal to 1 is supported
```

Comment (aspctu) [reply truss/config/trt_llm.py:78]: or 
```
max_beam_width > 1 is not supported
```

Comment (joostinyi) [reply truss/config/trt_llm.py:78]: Will change the copy in a follow up PR with similar CLI changes

**Pull request merged**